sparta/sparta/functional/Register.hpp

```diff
@@ -704,16 +704,16 @@ class RegisterBase : public TreeNode
      * marked as read-only regardless of any read-write fields containing those
      * bits
      */
-    RegisterBase(TreeNode *parent, const Definition *def)
+    RegisterBase(TreeNode *parent, const Definition def)
```

Comment:
@knute-sifive, in this constructor does it make more sense to reference `def` or `def_` in the body? I changed the Field and aliases initialization to use `def_` since they need a reference to it, but left the rest alone.

My question is really more regarding style/convention... looks strange to me to have both `def` and `def_` used in the body.